### PR TITLE
Restik unit test, e2e test

### PR DESCRIPTION
**Pull request by saumanbiswas**

### CODE REVIEW

File: client/clientset/fake/backup.go [hunk 1]
```diff
@@ -14,9 +14,9 @@ type FakeBackup struct {
 	ns   string
 }
 
-var certResource = schema.GroupVersionResource{Group: "appscode.com", Version: "v1beta1", Resource: "certificates"}
+var certResource = schema.GroupVersionResource{Group: "appscode.com", Version: "v1beta1", Resource: "backups"}
```
Comment (sadlil) [inline client/clientset/fake/backup.go]: Change the variable name.

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -587,7 +580,7 @@ func execLocal(s string) (string, error) {
 	return strings.TrimSuffix(string(cmdOut), "\n"), err
 }
 
-func getPasswordFromSecret(client *internalclientset.Clientset, secretName, namespace string) (string, error) {
+func getPasswordFromSecret(client *clientset.Clientset, secretName, namespace string) (string, error) {
```
Comment (sadlil) [inline pkg/controller/controller.go]: Isn't clientset.ClientSet is interface?

File: test/e2e/e2e.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	"github.com/appscode/restik/pkg/controller"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	"k8s.io/kubernetes/pkg/api"
+	"time"
+	"fmt"
+	"errors"
+)
+
+func runController() (*controller.Controller, error) {
+	config, err :=  clientcmd.BuildConfigFromFlags("", "/home/sauman/.kube/config")
```
Comment (sadlil) [inline test/e2e/e2e.go]: you can't point your pc specific config. What if i try to run those test in my machine?

File: test/e2e/e2e_test.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	fmt.Println("SUCCESS: ReplicationController Backup")
+
+	fmt.Println("\n**************************************\nCreating Replicaset -->", rs)
+	err = createReplicaset(watcher, rs, backupReplicaset)
+	if !assert.Nil(t, err) {
+		return
+	}
+	defer deleteReplicaset(watcher, rs)
+	time.Sleep(time.Second * 30)
+	fmt.Println("Starting backup for Replicaset...")
```
Comment (sadlil) [inline test/e2e/e2e_test.go]: all print should be done using `log` not fmt.Println()

File: test/e2e/kube_objects.go [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+import (
+	rapi "github.com/appscode/restik/api"
+	"github.com/appscode/restik/pkg/controller"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"log"
+)
+
+var namespace = "sauman"
```
Comment (sadlil) [inline test/e2e/kube_objects.go]: try creating a random.

```
var ns = Rand()
```

File: client/clientset/fake/extensions.go [hunk 6]
```diff
@@ -34,3 +35,8 @@ func NewFakeExtensionClient(objects ...runtime.Object) *FakeExtensionClient {
 func (m *FakeExtensionClient) Backups(ns string) client.BackupInterface {
 	return &FakeBackup{m.Fake, ns}
 }
+
+func (c *FakeExtensionClient) RESTClient() rest.Interface {
+	var ret *rest.RESTClient
+	return ret
```
Comment (sadlil) [inline client/clientset/fake/extensions.go]: You are not specifically setting the value. why not send `nil`?

File: pkg/controller/controller.go (hunk 2 above)
Comment (sadlil) [inline pkg/controller/controller.go]: use the interface. so this file can also be unit testable.

**Comment by TamalSaha** [comment]
@saumanbiswas add constants like below ad use them

``
const (
	ResourceKindBackup = "Backup"
	ResourceNameBackup = "backup"
	ResourceTypeBackup = "backups"
)
```

**Comment by sadlil** [comment]
LGTM

**Pull request merged**